### PR TITLE
UDN: Support managedTap binding, bump kubevirt-ipam-controller

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
Bump kubevirt-ipam-controller to v0.1.9-alpha
Adapt to support IPAM UDN tests with managedTap binding on CNAO.

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh [hunk 1]
```diff
@@ -53,6 +53,8 @@ main() {
     deploy_cnao
     deploy_cnao_cr
     ./hack/deploy-kubevirt.sh
+    ./cluster/kubectl.sh -n kubevirt patch kubevirt kubevirt --type=json --patch '[{"op":"add","path":"/spec/configuration/developerConfiguration","value":{"featureGates":[]}},{"op":"add","path":"/spec/configuration/developerConfiguration/featureGates/-","value":"NetworkBindingPlugins"}]'
```
Comment (RamLavi) [inline automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh:56]: please consider moving this to a function like `PatchManagedMacvtapNetworkBinding()`

**Review by RamLavi — COMMENTED** [review]
thanks for preparing this!

File: automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh (hunk 1 above)
Comment (oshoval) [reply automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh:56]: i dont think we need, it is just 2 lines and the whole file doesnt have funcs atm

Comment (RamLavi) [reply automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh:56]: It is not needed. it would just make it more readable. 
your call.

Comment (oshoval) [inline automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh:56]: TODO - add DynamicPodInterfaceNaming

**Comment by oshoval** [comment]
rebased

**Comment by oshoval** [comment]
added DynamicPodInterfaceNaming FG

File: automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh [hunk 2]
```diff
@@ -53,6 +53,8 @@ main() {
     deploy_cnao
     deploy_cnao_cr
     ./hack/deploy-kubevirt.sh
+    ./cluster/kubectl.sh -n kubevirt patch kubevirt kubevirt --type=json --patch '[{"op":"add","path":"/spec/configuration/developerConfiguration","value":{"featureGates":[]}},{"op":"add","path":"/spec/configuration/developerConfiguration/featureGates/-","value":"NetworkBindingPlugins"},{"op":"add","path":"/spec/configuration/developerConfiguration/featureGates/-","value":"DynamicPodInterfaceNaming"}]'
+    ./cluster/kubectl.sh -n kubevirt patch kubevirt kubevirt --type=json --patch '[{"op":"add","path":"/spec/configuration/network","value":{"binding":{"managedTap":{"domainAttachmentType":"managedTap","migration":{}}}}}]'
```
Comment (RamLavi) [inline automation/check-patch.e2e-kubevirt-ipam-controller-functests.sh:57]: I think it should be like this no?
```suggestion
    ./cluster/kubectl.sh -n kubevirt patch kubevirt kubevirt --type=json --patch '[{"op":"add","path":"/spec/configuration/network","value":{"binding":{"l2bridge":{"domainAttachmentType":"managedTap","migration":{}}}}}]'
```

**Comment by oshoval** [comment]
cherry-picked https://github.com/kubevirt/cluster-network-addons-operator/pull/1952 here 
lets see it pass
else would be glad if you can take it from here please @RamLavi

**Comment by RamLavi** [comment]
@oshoval I opened a [PR](https://github.com/kubevirt/cluster-network-addons-operator/pull/1954) to do the same. Please close this one

**Pull request closed without merging**